zhusuan/distributions/utils.py

```diff
@@ -182,26 +182,65 @@ def assert_same_float_and_int_dtype(tensors_with_name, dtype=None):
         raise TypeError("The argument 'dtype' must be in %s" % available_types)
 
 
-def assert_rank_at_least_one(tensor, name):
+def get_shape_list(tensor):
     """
-    Whether the rank of `tensor` is at least one.
+    :param tensor: A `tf.Tensor`.
+    :return: A list representing the shape of `tensor`, or None if the shape
+             is unknown. Each item in the list is either int or a `tf.Tensor`.
+    """
+    static_shape = tensor.get_shape()
+    if not static_shape:
+        return None
+    dynamic_shape = tf.shape(tensor)
+    ret = [(val or dynamic_shape[i])
+           for i, val in enumerate(static_shape.as_list())]
+    return ret
+
+
+def get_shape_at(tensor, axis):
+    """
+    Similar to `tf.shape(tensor)[axis]`, but return a constant when possible.
+
+    :param tensor: A `tf.Tensor`.
+    :param axis: `int`.
+    :return: The shape along the axis specified.
+    """
+    sizes_of_axes = get_shape_list(tensor)
+    if sizes_of_axes:
+        return sizes_of_axes[axis]
+    return tf.shape(tensor)[axis]
+
```

Comment:
I thought we only need tests for exportable functions (those in `__all__`). Should we add tests for the rest?

Reply:
But the exposed functions depend on these utils... only minor test that reflects the current use case should be good, also in case someone in future modifies these utils for their use, and the tests keep the original case safe.